ssl/quic/quic_txp.c

```diff
@@ -843,11 +843,39 @@ int ossl_quic_tx_packetiser_generate(OSSL_QUIC_TX_PACKETISER *txp,
             }
 
         if (first_el != QUIC_ENC_LEVEL_NUM
-            && total_dgram_size < min_dpl) {
+            && total_dgram_size < min_dpl
+            /*
+             * We might not be able to add padding, for example if we are using
+             * the ACK_ONLY archetype.
+             */
+            && pkt[first_el].geom.adata.allow_padding) {
```

Comment:
Why do we only check allow_padding for the first_el here? Couldn't we add the padding to any el if the first_el is not suitable?

Reply:
Yes &mdash; but in practice allow_padding is always the same for all ELs as the archetype is chosen for an entire datagram. So in practice it is not a real issue.